Feat: Make ERC20Issuance minting rights whitelist-based

0xNuggan opened the pull request:
## What has been done
- Substituted `address minter` with a mapping, so more than one address can mint.
- Expanded tests

0xNuggan committed:
make mininting whitelist

marvinkruse reviewed (approved):
lgtm

0xNuggan committed:
expand test and prepare  for next PR

marvinkruse committed:
Merge branch 'dev' into expandedERC20Issuance

marvinkruse committed:
Merge branch 'dev' into expandedERC20Issuance

Zitzak commented on src/modules/fundingManager/bondingCurve/tokens/ERC20Issuance_v1.sol:
```diff
@@ -92,9 +93,8 @@ contract ERC20Issuance_v1 is IERC20Issuance_v1, ERC20, Ownable {
 
     /// @notice Sets the address of the minter.
```
This function now serves for setting and removing. Maybe adjust the comment to reflect that change?

Zitzak replied:
Seeing that you also did that in the interface

Zitzak commented on test/external/ERC20Issuance.t.sol:
```diff
@@ -25,12 +27,22 @@ contract ERC20IssuanceTest is Test {
     │   └── It should revert
     └── When the caller is the Admin
-    └── It should set the new minter address
+    └── It should set the new minter address rights
```
```suggestion
        └── It should set the new minter address rights
```

Zitzak commented on test/external/ERC20Issuance.t.sol:
```diff
@@ -43,14 +55,42 @@ contract ERC20IssuanceTest is Test {
+        }
+
+        // state before
+        bool allowedBefore = token.allowedMinters(minter);
```
Unused variable

Zitzak reviewed (changes requested):
Reviewed this PR so I would understand the other one about the removing the deployment fromt he FM_BC init. Found some very minor stuff

0xNuggan committed:
use ERC20Capped

0xNuggan committed:
address review comments

0xNuggan committed:
move ERC20Issuance to external folder

0xNuggan committed:
Merge branch 'dev' into expandedERC20Issuance

Zitzak reviewed (approved):
LGTM. smart to make use of ERC20Capped as well as moving it to a different folder

marvinkruse committed:
Merge branch 'dev' into expandedERC20Issuance

marvinkruse reviewed (approved):
lgtm, under the condition that the tests are not failing anymore :D

0xNuggan committed:
[SC-700] Remove ERC20Issuance token deployment from FM_BC... contracts (#613)

* deploy Issuance token outside of Bonidng Curve initialization

* use ERC20Capped

* address review comments

* move ERC20Issuance to external folder

* Bug(SC-698): Rename function  and natspec of sellFor to sellTo (#611)

* Bug(SC-698): Rename function  and natspec of sellFor to sellTo

The function and natspec for sellFor were misleading, as we are not enabling selling for someone else, but sending the redeemed token to a specified address. The function name and comments have been updated to reflect this

* Fix: typo in natspec

---------

Co-authored-by: Marvin Kruse <marvinkruse@users.noreply.github.com>

* Test: Fix failing test with empty implementation (#614)

* CI: Unify actions into predependent job (#615)

* adapt tests after merge

---------

Co-authored-by: Marvin Kruse <marvinkruse@users.noreply.github.com>
Co-authored-by: Marvin <43185740+Zitzak@users.noreply.github.com>
